src/pages/Overview/index.js

```diff
@@ -1,16 +1,850 @@
+    // Stop propagation from going to the next page
+    e.stopPropagation();
+
+    const dropdownElem = document.getElementById(id);
+    if (dropdownElem.style.display === 'none') {
+      dropdownElem.style.display = 'block';
+    } else {
+      dropdownElem.style.display = 'none';
+    }
+  };
+
+  saveRuns = (e, rows) => {
+    // Stop propagation from going to the next page
+    if (e !== null) {
+      e.stopPropagation();
+    }
+    const { totalResultData } = this.state;
+    const { dispatch } = this.props;
+    const keys = rows.map(({ original }) => original.result);
+    keys.forEach(function(key) {
+      totalResultData.filter(item => item.result === key)[0].serverMetadata.dashboard.saved = true;
+    });
+    dispatch({
+      type: 'dashboard/updateResults',
+      payload: totalResultData,
+    }).then(() => {
+      this.setState({
+        totalResultData,
+      });
+      this.getSeparatedResults();
+    });
+  };
+
+  deleteResult = (e, rows) => {
+    // Stop propagation from going to the next page
+    if (e !== null) {
+      e.stopPropagation();
+    }
+    const { totalResultData } = this.state;
+    const { dispatch } = this.props;
+    const updatedResult = totalResultData.filter(item => !rows.includes(item.result));
+    dispatch({ type: 'dashboard/updateResults', payload: updatedResult })
+      .then(() => {
+        this.setState({ totalResultData: updatedResult });
+        this.getSeparatedResults();
+      })
+      .then(() => {
+        this.handleModalToggle(e, []);
+      })
+      .then(() => {
+        this.setState({ actionMessage: 'data succesfully deleted' });
+        setTimeout(() => {
+          this.setState({ actionMessage: '' });
+        }, 3000);
+      });
+  };
+
+  retrieveResults = row => {
+    const { dispatch } = this.props;
+    this.markResultSeen(row);
+    dispatch({
+      type: 'global/updateSelectedResults',
+      payload: [row],
+    }).then(() => {
+      dispatch(
+        routerRedux.push({
+          pathname: '/result',
+        })
+      );
+    });
+  };
+
+  navigateToExpiringResult = () => {
+    const { dispatch } = this.props;
+    dispatch(
+      routerRedux.push({
+        pathname: '/expiring-results',
+      })
+    );
+  };
+
+  fetchRunResult() {
+    const { dispatch, results, selectedDateRange, selectedControllers } = this.props;
+    if (Object.keys(results).length === 0) {
+      dispatch({
+        type: 'dashboard/fetchResults',
+        payload: {
+          selectedDateRange,
+          controller: selectedControllers,
+        },
+      }).then(() => this.getRunResult());
+    } else {
+      this.getRunResult();
+    }
+  }
+
+  handleChange() {
+    const { expiringData, newData } = this.state;
+    if (expiringData.length > 0) {
+      this.setState({
+        expiringData: [],
+      });
+    } else {
+      this.setState({
+        expiringData: newData.filter(x => getDiffDays(x.serverMetadata['server.deletion']) < 15),
+      });
+    }
+  }
```

Comment:
First, LGTM is flagging line 249 with the warning, "Component state update uses potentially inconsistent value."  It would be good to understand (and hopefully fix!) that before merging.

That noted, I'm having trouble with this code:  if I understand what I'm reading, then if there is expiring data then we ignore the `newData` and clear the existing expiring data; however, if there is no expiring data, then we set the expiring data based on the new data.  Is that right?  I would naively have thought that we would always set the expiring data based on the new data, and, if there is no new data, then we would clear the expiring data.

Finally, the `15` is a magic number which really should be defined as a symbolic constant/variable.  That will make it easy to change or to make it dynamic in the future.